packages/backend-storage/src/factory.ts

```diff
@@ -46,7 +55,139 @@ class AmplifyStorageFactory
         getInstanceProps
       );
     }
-    return constructContainer.getOrCompute(this.generator) as AmplifyStorage;
+    const amplifyStorage = constructContainer.getOrCompute(
+      this.generator
+    ) as AmplifyStorage;
+
+    const firstStorage = Stack.of(amplifyStorage).node.children.filter(
+      (el) => el instanceof AmplifyStorage
+    )[0] as AmplifyStorage;
+
+    /*
+     * only call Aspects once,
+     * otherwise there will be the an error -
+     * "there is already a construct with name 'storageRegion'"
+     */
+    if (firstStorage.name === this.props.name) {
+      Aspects.of(Stack.of(amplifyStorage)).add(
+        new StorageValidator(getInstanceProps.outputStorageStrategy)
+      );
+    }
+
+    return amplifyStorage;
+  };
+}
+
+/**
+ * StorageValidator class implements the IAspect interface.
+ */
+export class StorageValidator implements IAspect {
+  node: IConstruct;
+  outputStorageStrategy;
+  /**
+   * Constructs a new instance of the StorageValidator class.
+   */
+  constructor(
+    outputStorageStrategy: BackendOutputStorageStrategy<StorageOutput>
+  ) {
+    this.outputStorageStrategy = outputStorageStrategy;
+  }
+  /**
+   * Visit method to perform validation on the given node.
```

Comment:
please update comments to reflect new behaviors in this class.

Reply:
Done.